### PR TITLE
feat(bundler/NSIS): allow specifying custom lang files

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: tooling/bundler/src/bundle/windows/nsis.rs [hunk 1]
```diff
@@ -232,7 +228,28 @@ fn build_nsis_app_installer(
...
     }),
   );
+
+  let languages_data =  languages
+  .iter()
+  .filter_map(|lang| {
+    if let Some(data) = get_lang_data(lang, custom_language_files.as_ref()) {
+      Some(data)
+    } else {
+      log::warn!("Custom tauri messages for {lang} is not translated.\nIf it is a valid language listed on <https://github.com/kichik/nsis/tree/9465c08046f00ccb6eda985abbdbf52c275c6c4d/Contrib/Language%20files>, please open a Tauri feature request\n or you can provide a custom language file for it in `tauri.conf.json > tauri > bundle > windows > nsis > custom_language_files`");
```
Comment (FabianLars) [inline tooling/bundler/src/bundle/windows/nsis.rs]:
```suggestion
      log::warn!("Custom tauri messages for {lang} are not translated.\nIf it is a valid language listed on <https://github.com/kichik/nsis/tree/9465c08046f00ccb6eda985abbdbf52c275c6c4d/Contrib/Language%20files>, please open a Tauri feature request\n or you can provide a custom language file for it in `tauri.conf.json > tauri > bundle > windows > nsis > custom_language_files`");
```

File: core/tauri-utils/src/config.rs [hunk 2]
```diff
@@ -459,6 +459,13 @@ pub struct NsisConfig {
   ///
   /// See <https://github.com/kichik/nsis/tree/9465c08046f00ccb6eda985abbdbf52c275c6c4d/Contrib/Language%20files> for the complete list of languages.
   pub languages: Option<Vec<String>>,
+  /// An key-value pair where the key is the language and the
+  /// value is the path to a custom `.nsi` file that holds the translated text for tauri's custom messages.
+  ///
+  /// See <https://github.com/tauri-apps/tauri/blob/dev/tooling/bundler/src/bundle/windows/templates/nsis-languages/English.nsh> for an example `.nsi` file.
+  ///
+  /// **Note**: the key must be a valid NSIS language and it must be added to [`NsisConfig`]languages array,
+  pub custom_language_files: Option<HashMap<String, PathBuf>>,
```
Comment (FabianLars) [inline core/tauri-utils/src/config.rs]:
```suggestion
  /// A key-value pair where the key is the language and the
  /// value is the path to a custom `.nsh` file that holds the translated text for tauri's custom messages.
  ///
  /// See <https://github.com/tauri-apps/tauri/blob/dev/tooling/bundler/src/bundle/windows/templates/nsis-languages/English.nsh> for an example `.nsh` file.
  ///
  /// **Note**: the key must be a valid NSIS language and it must be added to [`NsisConfig`] languages array,
  pub custom_language_files: Option<HashMap<String, PathBuf>>,
```
Not 100% sure about the `::` rustdoc syntax thing.

Comment (amrbashir) [reply core/tauri-utils/src/config.rs]: yeah it doesn't work

**Comment by FabianLars** [comment]
(you need to rebuild the scheme after that one suggestion btw. oh and maybe copy the change to settings.rs too)

**Review by lucasfernog — DISMISSED** [review]

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**